12-solid/3-liskov-substitution/0-notes.md

```diff
@@ -0,0 +1,32 @@
+# Liskov substitution principle
+
+![Illustration: Liskov substitution principle](../storage/3-intro-liskov-substitution-principle.png)
+
+### "Let q(x) be a property provable about objects x of type T. Then q(y) should be provable for objects y of type S, where S is a subtype of T"
+
+* What it simply means is,
+  * If `B` is a sub-class of `A`, then objects of class `A` may be replaced with
+        objects of class `B`.
+  * Derived classes must be substitutable for their base classes.
+
+```php
+class A
+{
+    public function __invoke($foo)
+    {
+        // ...
+    }
+}
+
+class B extends A
+{
+    public function __invoke($foo)
```

Comment:
You can indicate its a overide, by phpdoc

Reply:
I put it in the body itself.. want to keep it simple here